### ISSUE TITLE
After ugrading Pillow 2.3.0 to 2.7.0 .eps-files are not recognized anymore

### ISSUE THREAD

**Issue by RomanTschirf** [python-pillow/Pillow#1104]
After updating to version 2.7.0 Pillow fails to recognize an input .eps-file. After downgrading to 2.3.0 again the file is read like intended.

The Error Message:

    Traceback (most recent call last):
       File "img_pil.py", line 25, in <module>
         img = Image.open('epstest.eps')
       File "build/bdist.linux-x86_64/egg/PIL/Image.py", line 2274, in open
    
    IOError: cannot identify image file 'epstest.eps'

Is this a bug?

**Comment by hugovk** [comment]
@RomanTschirf Can you reproduce it with any of these test eps files?

https://github.com/python-pillow/Pillow/tree/master/Tests/images

Or can you share a problematic file?

**Comment by hugovk** [comment]
And does it work with 2.4, 2.5 or 2.6?

**Comment by RomanTschirf** [comment]
@hugovk It worked with 2.3.0, 2.4.0 and 2.5.0 which is used now. 2.6.0 and 2.7.0 failed with the same error. I was using an 'old' (but simpler?) Illustrator 10-Eps format. I will test will the provided .eps-files.

**Comment by RomanTschirf** [comment]
All .eps-files (from those you proposed) failed when I tried to load them. Tested in 2.5.0 and 2.7.0.
```
Binary_preview_map.eps # The others failed the same way

Fails: 2.5.0:
Traceback (most recent call last): 
  File "img_pil.py", line 44, in <module> 
    img = open_eps(filename, dpi=300.0) 
  File "img_pil.py", line 30, in open_eps 
    img = Image.open(filename) 
  File "/usr/local/lib/python2.7/dist-packages/PIL/Image.py", line 2256, in open 
    % (filename if filename else fp)) 
IOError: cannot identify image file 'binary_preview_map.eps'

Fails: 2.7.0
Traceback (most recent call last):
  File "img_pil.py", line 44, in <module>
    img = open_eps(filename, dpi=300.0)
  File "img_pil.py", line 30, in open_eps
    img = Image.open(filename)
  File "/usr/local/lib/python2.7/dist-packages/PIL/Image.py", line 2274, in open
    % (filename if filename else fp))
IOError: cannot identify image file 'binary_preview_map.eps'
```
I uploaded some .eps-files the way I would use them plus the python-file to
https://dl.dropboxusercontent.com/u/45390772/eps-files-for-testing.tar.gz

Also, when trying to open your eps-files with Illustrator or Photoshop (both CS6) they were not able to interpret them.

**Comment by wiredfool** [comment]
@RomanTschirf What platform are you on? The only think that I can think of that's changed that could affect this is that the handling of input streams and line endings changed, and that's a notorious place where platforms can be different.

**Comment by RomanTschirf** [comment]
My .eps-files are produced unter Windows7,  Development under Ubuntu 14.04. Tried to test under Windows also, but getting ghostscript to run is a pain (and I`m not a pro..)

**Comment by cgohlke** [comment]
Looks like the EPS parser tries to index blank lines. Try:

```diff
diff --git a/PIL/EpsImagePlugin.py b/PIL/EpsImagePlugin.py
index d077ca1..83024b6 100644
--- a/PIL/EpsImagePlugin.py
+++ b/PIL/EpsImagePlugin.py
@@ -275,13 +275,13 @@ class EpsImageFile(ImageFile.ImageFile):

             s = fp.readline().strip('\r\n')

-            if s[0] != "%":
+            if s[:1] != "%":
                 break

         #
         # Scan for an "ImageData" descriptor

-        while s[0] == "%":
+        while s[:1] == "%":

             if len(s) > 255:
                 raise SyntaxError("not an EPS file")
```

**Comment by RomanTschirf** [comment]
@cgohlke Thanks! I´m only able to test in the upcoming days - I will keep you updated.

**Comment by hugovk** [comment]
@RomanTschirf Any luck?

**Comment by aclark4life** [comment]
@RomanTschirf it would be great to get a fix for this in 2.8.0!

**Comment by hugovk** [comment]
This can be reproduced on Travis CI.

---

Here's a failing build with a simplified test case based on https://github.com/python-pillow/Pillow/issues/1104#issuecomment-73365470:

```python
    def test_open_eps(self):
        # https://github.com/python-pillow/Pillow/issues/1104
        # Arrange
        FILES = ["Tests/images/illu10_no_preview.eps",
                 "Tests/images/illu10_preview.eps",
                 "Tests/images/illuCS6_no_preview.eps",
                 "Tests/images/illuCS6_preview.eps"]

        # Act
        for filename in FILES:
            img = Image.open(filename)

        # Assert
        self.assertEqual(img.mode, "RGB")
```

```
ERROR: TestFileEps.test_open_eps
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/hugovk/Pillow/Tests/test_file_eps.py", line 236, in test_open_eps
    img = Image.open(filename)
  File "/home/travis/build/hugovk/Pillow/PIL/Image.py", line 2284, in open
    % (filename if filename else fp))
IOError: cannot identify image file 'Tests/images/illu10_no_preview.eps'
```
https://travis-ci.org/hugovk/Pillow/builds/55938972

---

And here's a passing build with this test and @cgohlke's fix:

https://travis-ci.org/hugovk/Pillow/builds/55939236

---

PR created here: https://github.com/python-pillow/Pillow/pull/1152

@RomanTschirf Is it ok to distribute those test images under a MIT-style licence?
